File: 6_kyu/Braking_well.jl
module Braking
    export dist, speed
    
    const EARTH_GRAVITY = 9.81
    
    function dist(v::Number, μ::Number)::Real
        # convert from km/h to m/s
        v /= 3.6
        # Reaction time: 1
        v + v^2/2/μ/EARTH_GRAVITY
    end

    # Reaction time: 1
    function speed(d::Number, μ::Number)::Real
        μg = μ * EARTH_GRAVITY
        v = sqrt( μg^2 + 2d*μg ) - μg
        # Convert m/s in km/h
        3.6v
    end
end

File: 6_kyu/Braking_well.go
package kata

import "math"

// Gravity acceleration at earth surface
const earth_gravity float64 = 9.81

// Factor to convert km/h into m/s
const ms2kmh float64 = 3.6

func Dist(v, mu float64) float64 {
	// convert from km/h to m/s
	v /= ms2kmh
	// Reaction time: 1 second
	return v + v*v/2/mu/earth_gravity
}

func Speed(d, mu float64) float64 {
	mug := mu * earth_gravity
	v := (math.Sqrt(1+2*d/mug) - 1) * mug
	// Convert m/s in km/h
	return v * ms2kmh
}
